packages/databyss-ui/modules/SourcesValueList/SourceModal.js

```diff
@@ -0,0 +1,46 @@
+import React, { useState } from 'react'
+import { useSourceContext } from '@databyss-org/services/sources/SourceProvider'
+import { ModalWindow } from '@databyss-org/ui/primitives'
+import { hideModal } from '@databyss-org/ui/components/Navigation/NavigationProvider/actions'
+import { useNavigationContext } from '@databyss-org/ui/components/Navigation/NavigationProvider/NavigationProvider'
+import SourcesValueList from './SourcesValueList'
+
+const SourceModal = ({ sourceId, visible, onUpdateSource }) => {
+  const { setSource } = useSourceContext()
+  const [values, setValues] = useState(null)
+  const [, dispatchNav] = useNavigationContext()
+
+  const onChange = _values => {
```

Comment:
Unnecessary function, we can just use `setValues` directly

Reply:
ill keep this function until we discuss the best way to update the source state without causing a re-render.